db/view/view.cc

```diff
@@ -2549,38 +2549,45 @@ void view_builder::on_drop_view(const sstring& ks_name, const sstring& view_name
 }
 
 future<> view_builder::do_build_step() {
-    return seastar::async([this] {
-        exponential_backoff_retry r(1s, 1min);
-        while (!_base_to_build_step.empty() && !_as.abort_requested()) {
-            auto units = get_units(_sem, 1).get();
-            ++_stats.steps_performed;
-            try {
-                execute(_current_step->second, exponential_backoff_retry(1s, 1min));
-                r.reset();
-            } catch (const abort_requested_exception&) {
-                return;
-            } catch (...) {
-                ++_current_step->second.base->cf_stats()->view_building_paused;
-                ++_stats.steps_failed;
-                auto base = _current_step->second.base->schema();
-                vlogger.warn("Error executing build step for base {}.{}: {}", base->ks_name(), base->cf_name(), std::current_exception());
-                r.retry(_as).get();
-                initialize_reader_at_current_token(_current_step->second).get();
-            }
-            if (_current_step->second.build_status.empty()) {
-                auto base = _current_step->second.base->schema();
-                auto reader = std::move(_current_step->second.reader);
-                _current_step = _base_to_build_step.erase(_current_step);
-                reader.close().get();
-            } else {
-                ++_current_step;
-            }
-            if (_current_step == _base_to_build_step.end()) {
-                _current_step = _base_to_build_step.begin();
+    // View building is initiated during a schema merge where the new view is added.
+    // The schema merge is performed in the gossip scheduling group, but the actual
+    // work of building the view, which may be time-consuming, is done in the streaming
+    // (maintenance) scheduling group, so that it doesn't impact the processing of tasks
+    // in the gossip scheduling group.
```

Comment:
I think this comment is prone to getting out of date. It describes how do_build_step is used, but you've put it in the implementation of the function, so it will be very easy to somebody to forget to update it later. In particular, I imagine that this function, or some its derivative, will be invoked by the view build coordinator in the future.

In general, I think this comment is not needed - it's not important with what scheduling group this function is called, we just care that it runs in the scheduling group we want. You can explain why we want to run it in the streaming scheduling group, but I don't think that more than that is needed.

Reply:
Ok, I removed most of the comment

Reply:
@piodul please re-review and approve or requests further changes.